sys/include/net/sock/ip.h

```diff
@@ -0,0 +1,204 @@
+ *          ((local->netif != SOCK_ADDR_ANY_NETIF) ||
+ *          (remote->netif != SOCK_ADDR_ANY_NETIF))` if neither is `NULL`).
+ * @return  -ENOMEM, if the stack can't provide enough resources for `sock` to
+ *          be created.
+ * @return  -EPROTONOSUPPORT, if `local != NULL` or `remote != NULL` and
+ *          proto is not supported by sock_ip_ep_t::family of @p local or @p
+ *          remote.
+ */
+int sock_ip_create(sock_ip_t *sock, const sock_ip_ep_t *local,
+                   const sock_ip_ep_t *remote, uint8_t proto, uint16_t flags);
+
+/**
+ * @brief   Closes a raw IPv4/IPv6 sock object
+ *
+ * @pre `(sock != NULL)`
+ *
+ * @param[in] sock  A raw IPv4/IPv6 sock object.
+ */
+void sock_ip_close(sock_ip_t *sock);
+
+/**
+ * @brief   Gets the local end point of a raw IPv4/IPv6 sock object
+ *
+ * @pre `(sock != NULL) && (ep != NULL)`
+ *
+ * @param[in] sock  A raw IPv4/IPv6 sock object.
+ * @param[out] ep   The local end point.
+ *
+ * @return  0 on success.
+ * @return  -EADDRNOTAVAIL, when @p sock has no end point bound to it.
+ */
+int sock_ip_get_local(sock_ip_t *sock, sock_ip_ep_t *ep);
+
+/**
+ * @brief   Gets the remote end point of a UDP sock object
+ *
+ * @pre `(sock != NULL) && (ep != NULL)`
+ *
+ * @param[in] sock  A UDP sock object.
+ * @param[out] ep   The remote end point.
+ *
+ * @return  0 on success.
+ * @return  -ENOTCONN, when @p sock has no remote end point bound to it.
+ */
+int sock_ip_get_remote(sock_ip_t *sock, sock_ip_ep_t *ep);
+
+/**
+ * @brief   Receives a message over IPv4/IPv6 from remote end point
+ *
+ * @pre `(sock != NULL) && (data != NULL) && (max_len > 0)`
+ *
+ * @param[in] sock      A raw IPv4/IPv6 sock object.
+ * @param[out] data     Pointer where the received data should be stored.
+ * @param[in] max_len   Maximum space available at @p data.
+ *                      If received data exceeds @p max_len the data is
+ *                      truncated and the remaining data can be retrieved
+ *                      later on.
+ * @param[in] timeout   Timeout for receive in microseconds.
+ *                      This value can be ignored (no timeout) if the
+ *                      @ref sys_xtimer module is not present and the stack does
+ *                      not support timeouts on its own.
+ *                      May be 0 for no timeout.
+ * @param[out] remote   Remote end point of the received data.
+ *                      May be NULL, if it is not required by the application.
+ *
+ * @note    Function blocks if no packet is currently waiting.
+ *
+ * @return  The number of bytes received on success.
+ * @return  0, if no received data is available, but everything is in order.
+ * @return  -EADDRNOTAVAIL, if local of @p sock is not given.
+ * @return  -ENOBUFS, if buffer space is not large enough to store received
+ *          data.
+ * @return  -ENOMEM, if no memory was available to receive @p data.
+ * @return  -EPROTO, if source address of received packet did not equal
```

Comment:
This is not a doc discussion, though but a conceptional discussion of that you said there are non any more ;-)